machine_test.go

```diff
@@ -252,7 +254,15 @@ func testAttachVsock(ctx context.Context, t *testing.T, m *Machine) {
 func testStartInstance(ctx context.Context, t *testing.T, m *Machine) {
 	err := m.startInstance(ctx)
 	if err != nil {
-		t.Errorf("startInstance failed: %s", err)
+		if syncErr, ok := err.(*operations.CreateSyncActionDefault); ok &&
+			strings.HasPrefix(syncErr.Payload.FaultMessage, "Cannot create vsock device") {
```

Comment:
This seems a little fragile since this message is coming back from `firecracker` and not the SDK. I am a little worried that `firecracker` may update this message which would cause the SDK to start failing. Even if we fixed it, users using the old SDK but newer `firecracker` would have failed builds. I would much prefer trying to validate this in the SDK or maybe wrapping the error received from `firecracker`? Don't know if I like the wrapping idea either

Reply:
Yeah, it's definitely fragile.  Unfortunately the errors in the Firecracker API are not modeled any more than just a string.  I opted to put this in the test instead of the SDK itself precisely because I can't guarantee that they won't change; I don't want to guarantee that the SDK functions if I'm relying on strings.  I want to fix it here so that anyone trying to build the SDK can validate what they're seeing, and then propose that we add explicit error modeling to the Firecracker API.

Reply:
Alright, that sounds like a plan! :shipit:

Reply:
This is the current API model:

https://github.com/firecracker-microvm/firecracker-go-sdk/blob/ec0a04065cf1307665ea30a1ae82b9a14da54401/client/swagger.yaml#L396-L400